src/Watch.php

```diff
@@ -30,7 +30,7 @@ public function __construct(Source $src, array $tasks)
                             'yellow'
                         );
                         $distFile->setLastChangeTime($timeChange);
-                        Phulp::start($tasks);
+                        $callback();
```

Comment:
We have to design a clear architecture.

In your original code, you defined that the `Phulp` class has a `watch()` method that in turn creates a new `Watch` object (and by constructing, the watch task is [already] triggered). In that watch task (i.e., inside the constructor), you referenced the (static) method `start()`.
And this is strange. P creates a new W, that references P and starts stuff.

I refactored this in a way that we now pass a callback (currently, a closure) to W, and W only executes the callback - without having a real reference to (and thus dependency on) Phulp.

Regarding your question:
> Should this callback receive by parameter the `$phulp`?

No, it should not. Because the callback already executes a specific method, `start()`, on the Phulp object.

I'm not yet really satisfied with the architecture. But it is (a little bit) more separated than before, and it works. For now.